backend/chromeApiServer/src/main/java/com/canvas/service/SubmissionDirectoryService.java

```diff
@@ -105,6 +106,23 @@ private SubmissionFile[] writeSubmissionFiles(Map<String, byte[]> submissionFile
         return submissionFiles.toArray(new SubmissionFile[0]);
     }
 
+    public ResponseEntity<Deletion> deleteSubmissionDirectory(ExtensionUser user) {
+        String submissionDirectory = generateUniqueDirectoryName(user.getCourseId(), user.getAssignmentId(), user.getStudentId());
+        String description = user.getCourseId() + " " +  user.getAssignmentId() + " " + user.getStudentId() + " " + submissionDirectory;
+
+        // Now delete
+        fileService.deleteDirectory(submissionDirectory);
+
+        // TODO: FileService.deleteDirectory signature could be changed to boolean and plugged into success for builder
+        // if successful deletion
+        Deletion deletion = Deletion.builder()
+                .success(true)
```

Comment:
can we add a case where the deletion is unsuccessful such as if the directory doesn't exist, and send back an error response?

Reply:
so in order to do this, from what I understand in the code and mentioned in the TODO above, we would have to change method signature of FileService.deleteDirectory from void too boolean. I don't think that will break anything but I will try and include it in some tests